modules/workflow-service-impl/src/main/java/org/opencastproject/workflow/impl/WorkflowServiceImpl.java

```diff
@@ -2198,8 +2198,24 @@ public void repopulate() throws IndexRebuildException {
               }
               current++;
 
-              var updatedWorkflowData = index.getEvent(indexData.getMediaPackageId(), indexData.getOrganizationId(),
-                        securityService.getUser());
+              String orgid = indexData.getOrganizationId();
+              if (null == orgid) {
+                //We're assuming here that mediapackages don't change orgs
+                orgid = assetManager.getSnapshotsById(indexData.getMediaPackageId())
+                    .getSnapshots().head2().getOrganizationId();
+                //We try-catch here since it's possible for the WF to exist in the *index* but not in the *DB*
+                // It probably shouldn't be, but that won't keep it from happening anyway.
+                try {
+                  //NB: This version of getWorkflow takes the org id, which in this case is null
+                  // Using the normal version filters by org, and since this workflow has a NULL org it can't be found
+                  WorkflowInstance instance = persistence.getWorkflow(indexData.getId(), null);
+                  instance.setOrganizationId(orgid);
+                  persistence.updateInDatabase(instance);
+                } catch (NotFoundException e) {
+                  logger.warn("Workflow {} not found in the database, but present in the index", indexData.getId());
```

Comment:
I don't think this can ever happen, because the info that this workflow exists came from the database, not the index (even though it's called indexData).

Reply:
Ok, will remove the logging line.  `getWorkflow` throws this so we have to catch it eventually :)